lib/tbd/framedivider.rb

```diff
@@ -542,9 +549,10 @@ def openings(model, surface)
     #
     four = (s.vertices.size == 4)
 
-    if s.windowPropertyFrameAndDivider.empty?
+    if tubular || s.windowPropertyFrameAndDivider.empty?
       vec = s.vertices
       area = gross
+      n = s.outwardNormal if unhinged
```

Comment:
1. TDDs can't have _frame & divider_ components.
2. Can't rely on parent surface normal for _unhinged_ TDD domes.